src/msg/async/AsyncMessenger.cc

```diff
@@ -513,20 +264,29 @@ class C_handle_reap : public EventCallback {
 AsyncMessenger::AsyncMessenger(CephContext *cct, entity_name_t name,
                                string mname, uint64_t _nonce, uint64_t features)
   : SimplePolicyMessenger(cct, name,mname, _nonce),
-    processor(this, cct, _nonce),
     dispatch_queue(cct, this, mname),
     lock("AsyncMessenger::lock"),
     nonce(_nonce), need_addr(true), did_bind(false),
     global_seq(0), deleted_lock("AsyncMessenger::deleted_lock"),
     cluster_protocol(0), stopped(true)
 {
   ceph_spin_init(&global_seq_lock);
-  cct->lookup_or_create_singleton_object<WorkerPool>(pool, WorkerPool::name);
-  local_worker = pool->get_worker();
-  local_connection = new AsyncConnection(cct, this, &dispatch_queue, local_worker);
+  static const string uniq_name = "AsyncMessenger::NetworkStack";
+  StackSingleton *single;
+  cct->lookup_or_create_singleton_object<StackSingleton>(single, uniq_name);
+  stack = single->stack;
+  stack->start();
```

Comment:
can we postpone `start()` until we really need the messenger to transmit messages?